datafusion/core/src/datasource/datasource.rs

```diff
@@ -72,3 +72,9 @@ pub trait TableProvider: Sync + Send {
         Ok(TableProviderFilterPushDown::Unsupported)
     }
 }
+
+/// A factory which creates TableProviders given appropriate information
```

Comment:
```suggestion
/// A factory which creates [`TableProvider`]s at runtime given a URL.
/// 
/// For example, this can be used to create a table "on the fly" 
/// from a directory of files only when that name is referenced.  
```